contract/src/main.rs

```diff
@@ -16,193 +19,139 @@ use types::{
+        String::from("read"),
+        vec![
+            Parameter::new("uref", CLType::String),
+            Parameter::new("dict_key", CLType::String),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-    entry_points.add_entry_point(endpoint(
-        "store_tuple3",
-        CLType::Tuple3([
-            Box::new(CLType::PublicKey),
-            Box::new(CLType::Option(Box::new(CLType::String))),
-            Box::new(CLType::U512),
-        ]),
+    entry_points.add_entry_point(EntryPoint::new(
+        String::from("update"),
+        vec![
+            Parameter::new("uref", CLType::String),
+            Parameter::new("dict_key", CLType::String),
+            Parameter::new("dict_value", CLType::Option(Box::new(CLType::String))),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-    entry_points.add_entry_point(endpoint(
-        "store_tuple2",
-        CLType::Tuple2([Box::new(CLType::String), Box::new(CLType::U512)]),
+    entry_points.add_entry_point(EntryPoint::new(
+        String::from("delete"),
+        vec![Parameter::new("uref", CLType::String)],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-
     let (contract_hash, _) =
         storage::add_contract_version(contract_package_hash, entry_points, Default::default());
-    runtime::put_key("kvstorage_contract", contract_hash.into());
+    runtime::put_key("kvstorage_contract_hash", contract_hash.into());
     let contract_hash_pack = storage::new_uref(contract_hash);
-    runtime::put_key("kvstorage_contract_hash", contract_hash_pack.into())
+    runtime::put_key("kvstorage_contract_hash_wrapped", contract_hash_pack.into())
 }
 
-fn read_and_store<T: CLTyped + FromBytes + ToBytes>() {
-    let name: String = runtime::get_named_arg("name");
-    let value: T = runtime::get_named_arg("value");
-    set_key(name.as_str(), value);
+fn create_key<T: ToBytes + CLTyped>(name: &str, dict_name: &str, dict_value: T) {
+    match runtime::get_key(name) {
+        None => {
+            // storage::new_dictionary function creates new URef that represents a seed for a dictionary partition
+            // of the global state and puts it under named keys.
+            // Any value can be stored directly under named keys using storage::new_uref function.
+            // But for more organisational structure, it's better to store values under dictionary_item_key in the
+            // dictionary and control this dictionary with one dictionary_seed_uref.
+            // It's possible to have multiple dictionaries for big data structure.
+            let uref = storage::new_dictionary(dict_name).unwrap_or_revert();
+            storage::dictionary_put(uref, dict_name, dict_value);
+            runtime::put_key(name, uref.into());
+        }
+        Some(_) => {
+            runtime::revert(ApiError::None);
```

Comment:
Having Some(_) shouldn't return Error::None. Just overall use your own error types instead of the network ones to make discerning errors easier.